Test/dafny0/DefiniteAssignment.dfy

```diff
@@ -77,12 +82,12 @@ method Main()
                     // previous assertion says, y were null
 }
 
-method DontForgetHavoc<G>(a: G, h: int) returns (k: G) {
+method DontForgetHavoc<G(00)>(a: G, h: int) returns (k: G) {
   var x:G, y:G, z:G := a, *, a;
   if h < 10 {
     k := x;  // fine
   } else if h < 20 {
-    k := y;  // error: y has not yet been assigned
+    k := y;
   } else {
     z := *;
     return z;  // this is fine, since z was assigned before the havoc
```

Comment:
Why does 'z was assigned before the havoc' matter? It doesn't for 'y', right?

Reply:
I think this comment can go